### PR TITLE
Bug: Read override config file in heron executor

### PR DESCRIPTION
**Pull request by rohanag12**
Heron executor should read config from `override.yaml` when connecting to the statemgr. More details in #3287.

### CODE REVIEW

**Comment by rohanag12** [comment]
@nwangtw @nlu90

File: heron/executor/src/python/heron_executor.py [hunk 1]
```diff
@@ -1109,10 +1109,15 @@ def start_state_manager_watches(self):
     Receive updates to the packing plan from the statemgrs and update processes as needed.
     """
     Log.info("Start state manager watches")
+
+    with open(self.override_config_file, 'r') as stream:
+      overrides = yaml.load(stream)
+    overrides["heron.statemgr.connection.string"] = self.state_manager_connection
```
Comment (xiaoyao1991) [inline heron/executor/src/python/heron_executor.py]: Can `yaml.load` fail in any case? If so, should we check null and initialize `overrides` if necessary?

Comment (rohanag12) [reply heron/executor/src/python/heron_executor.py]: You're right, if the file is empty, `yaml.load` returns `None`. Added a check, thanks!

**Comment by xiaoyao1991** [comment]
👍

**Pull request merged**